lib/ipam/ipam_block.go

```diff
@@ -146,14 +173,15 @@ func getHostAffinity(block *model.AllocationBlock) string {
 	return ""
 }
 
-func (b allocationBlock) numFreeAddresses() int {
+func (b allocationBlock) NumFreeAddresses() int {
 	return len(b.Unallocated)
 }
 
-// empty returns true if the block has released all of its assignable addresses,
-// and returns false if any assignable addresses are in use.
-func (b allocationBlock) empty() bool {
-	return b.containsOnlyReservedIPs()
+func (b allocationBlock) empty(windowsHost bool) bool {
```

Comment:
Do we need this change? I thought the old approach worked for both windows and linux hosts.

Reply:
The code is same as private master and v3.12. I think we need this change.

Reply:
Nah, I'm pretty sure those branches are wrong. Lets talk about it.

Reply:
Reverted to OS version